docs/testplan/transceiver_onboarding/cmis_cdb_firmware_upgrade_testplan.md

```diff
@@ -0,0 +1,489 @@
+# CMIS CDB Firmware Upgrade Test Plan
+
+## Scope
+
+This test plan outlines a comprehensive framework for testing the firmware upgrade/downgrade of CMIS compliant transceivers being onboarded to SONiC. The goal is to automate all tests listed in this document.
+
+**Optics Scope**:
+The test plan includes various optics types, such as:
+
+- Active Optical Cables (AOC)
+- Active Electrical Cables (AEC)
+- DR8 optics
+- Direct Attach Cables (DAC)
+- Short Range/Long Range (SR/LR) optics
+- SONiC-supported breakout cables
+
+**Optics Specifications**:
+Tests will cover optics compliant with:
+
+- CMIS
+- C-CMIS
+
+## Testbed Topology
```

Comment:
@az-pz We don't need to duplicate this here since this is already specified in the transciever onboarding doc. you can reference that here.